core/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -44,9 +44,20 @@
  */
 public final class RepositoryData implements ToXContent {
 
-    public static final RepositoryData EMPTY = new RepositoryData(Collections.emptyList(), Collections.emptyMap());
+    /**
+     * The generation value indicating the repository has no index generational files.
+     */
+    public static final long EMPTY_REPO_GEN = -1L;
+    /**
+     * An instance initialized for an empty repository.
+     */
+    public static final RepositoryData EMPTY = new RepositoryData(EMPTY_REPO_GEN, Collections.emptyList(), Collections.emptyMap());
 
     /**
+     * The generational id of the index file from which the repository data was read.
+     */
+    private final long genId;
```

Comment:
Discussed this with @abeyad more. It looks like it should be part of RepositoryData after all, otherwise the index generation abstractions is getting exposed on the Repository interface layer, where it makes even less sense.